File: src/main/scala/com/equilcraft/bigsteppa/common/gui/ArmorStandGuiHandler.java
package com.equilcraft.bigsteppa.common.gui;

import com.equilcraft.bigsteppa.client.gui.GuiArmorStand;
import com.equilcraft.bigsteppa.common.inventory.ContainerArmorStand;
import com.equilcraft.bigsteppa.common.tile.TileArmorStand;
import cpw.mods.fml.common.network.IGuiHandler;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.world.World;

public class ArmorStandGuiHandler implements IGuiHandler {

    public static final int GUI_ID = 1;

    @Override
    public Object getServerGuiElement(
        int id,
        EntityPlayer player,
        World world,
        int x,
        int y,
        int z
    ) {
        TileEntity tile = world.getTileEntity(x, y, z);
        return id == GUI_ID && tile instanceof TileArmorStand
            ? new ContainerArmorStand(player.inventory, (TileArmorStand) tile)
            : null;
    }

    @Override
    public Object getClientGuiElement(
        int id,
        EntityPlayer player,
        World world,
        int x,
        int y,
        int z
    ) {
        TileEntity tile = world.getTileEntity(x, y, z);
        return id == GUI_ID && tile instanceof TileArmorStand
            ? new GuiArmorStand(player.inventory, (TileArmorStand) tile)
            : null;
    }
}

File: src/main/scala/com/equilcraft/bigsteppa/common/gui/ArmorStandGuiHandler.scala
package com.equilcraft.bigsteppa.common.gui

import com.equilcraft.bigsteppa.client.gui.armorstand.GuiArmorStand
import com.equilcraft.bigsteppa.common.inventory.armorstand.ContainerArmorStand
import com.equilcraft.bigsteppa.common.tile.armorstand.TileArmorStand
import cpw.mods.fml.common.network.IGuiHandler
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.world.World

class ArmorStandGuiHandler extends IGuiHandler {
  override def getServerGuiElement(
    id: Int,
    player: EntityPlayer,
    world: World,
    x: Int,
    y: Int,
    z: Int
  ): AnyRef =
    world.getTileEntity(x, y, z) match {
      case armorStand: TileArmorStand if id == ArmorStandGuiHandler.GuiId =>
        new ContainerArmorStand(player.inventory, armorStand)
      case _ => null
    }

  override def getClientGuiElement(
    id: Int,
    player: EntityPlayer,
    world: World,
    x: Int,
    y: Int,
    z: Int
  ): AnyRef =
    world.getTileEntity(x, y, z) match {
      case armorStand: TileArmorStand if id == ArmorStandGuiHandler.GuiId =>
        new GuiArmorStand(player.inventory, armorStand)
      case _ => null
    }
}

object ArmorStandGuiHandler {
  final val GuiId = 1
}
